internal/buildengine/engine.go

```diff
@@ -1414,6 +1414,9 @@ func (e *Engine) tryDeployFromQueue(ctx context.Context, deployment *pendingDepl
 	modules := map[string]bool{}
 	for _, module := range deployment.modules {
 		modules[module.Config.Module] = true
+		for _, dep := range module.dependencies {
+			modules[dep] = true
```

Comment:
Do we need to be recursive?
If C depends on B depends on A... A is in an active changeset, B is waiting, C finishes building... it should also wait

Reply:
I don't think so, although maybe we need to also check the build queue to see if direct dependencies are queued up. If the A and C changes are unrelated then this is fine, but if the sequence was A triggers B which triggers C then we want C to wait because B is in the queue.

Reply:
makes sense to me